### PR TITLE
feat: Sort/Import tokens 🚧 

### PR DESCRIPTION
**Pull request by gambinish**
🚧 In progress, plz don't merge 🚧 

## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27084?quickstart=1)

## **Related issues**

Fixes:

## **Screenshots/Recordings**

https://github.com/user-attachments/assets/9e0d04ee-e8d7-474c-93f8-7adccbe2a470

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/assets/asset-list/asset-list-control-bar/asset-list-control-bar.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            token.symbol, // tokenSymbol
+            false, // no currency symbol prefix
+            false, // no ticker symbol suffix
+          ) || '0';
+      });
+
+      setTokenList(tokensWithBalances);
+      setLoading(loading);
+    }
+  }, [accountTotalFiatBalance]);
```
Comment (gambinish) [inline ui/components/app/assets/asset-list/asset-list-control-bar/asset-list-control-bar.tsx:113]: I'm probably going to break this out into it's own hook

File: ui/components/app/assets/asset-list/sort-control/sort-control.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      {isSelected && (
+        <Box
+          className="selectable-list-item__selected-indicator"
+          borderRadius={BorderRadius.pill}
+          backgroundColor={BackgroundColor.primaryDefault}
+        />
+      )}
+    </Box>
+  );
+};
```
Comment (gambinish) [inline ui/components/app/assets/asset-list/sort-control/sort-control.tsx:44]: @georgewrmarshall I'm tagging you here because we talked about potentially breaking this out into the component library. Not sure if we want to do that here or in a different PR.

File: ui/components/app/assets/asset-list/asset-list-control-bar/index.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.asset-list-control-bar {
+  display: flex;
+  justify-content: space-between;
```
Comment (georgewrmarshall) [inline ui/components/app/assets/asset-list/asset-list-control-bar/index.scss:3]: Can you use style utility props instead of styles here?

File: ui/components/app/assets/asset-list/asset-list-control-bar/index.scss [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  cursor: pointer;
+  padding-top: 16px;
+  margin-left: 16px;
+  margin-right: 16px;
```
Comment (georgewrmarshall) [inline ui/components/app/assets/asset-list/asset-list-control-bar/index.scss:7]: I'm not sure if there's a formal name for this, but I call adding margins/padding at the root level of a UI element "implied margin/padding," where the spacing is inferred from the parent. Outside of the parent, these margins and paddings don't have much meaning, so I prefer to allow margins and paddings to be applied via props during the component's placement within the parent, for example:

```
<AssetListControlBar marginLeft={16} marginRight={16} paddingTop={16} />
```

It's also fine to set defaults that can be overridden. In the root `Box` element, we might do something like:

```
<Box className="asset-list-control-bar" ref={controlBarRef} marginLeft={16} marginRight={16} paddingTop={16} {...props} />
```

This allows the component to be used in other places without carrying the implied margin/padding with it.

File: ui/components/app/assets/asset-list/asset-list-control-bar/index.scss [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+  button:hover {
+    background-color: var(--color-background-hover);
+  }
```
Comment (georgewrmarshall) [inline ui/components/app/assets/asset-list/asset-list-control-bar/index.scss:11]: Could we use a classname here instead of the element? This will help with discoverability and reduce any issues if we were to use another button element within this component

**Review by georgewrmarshall — COMMENTED** [review]
This is looking great! Not sure if it was ready for review but it was in my notifications so apologies if it was premature!

**Comment by gambinish** [comment]
> This is looking great! Not sure if it was ready for review but it was in my notifications so apologies if it was premature!

Appreciate the review! I'll implement your feedback shortly

File: ui/components/app/assets/asset-list/sort-control/sort-control.tsx (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/app/assets/asset-list/sort-control/sort-control.tsx:44]: A different PR would be great but I don't want it to halt progress, I could help you get it in a state tomorrow if that works!

File: ui/components/app/assets/asset-list/asset-list-control-bar/index.scss (hunk 3 above)
Comment (gambinish) [reply ui/components/app/assets/asset-list/asset-list-control-bar/index.scss]: https://github.com/MetaMask/metamask-extension/pull/27084/commits/4baa9ad7f67c1daf8b359232dd7dbbe3902d84a7

File: ui/components/app/assets/asset-list/asset-list-control-bar/index.scss (hunk 5 above)
Comment (gambinish) [reply ui/components/app/assets/asset-list/asset-list-control-bar/index.scss]: https://github.com/MetaMask/metamask-extension/pull/27084/commits/4baa9ad7f67c1daf8b359232dd7dbbe3902d84a7

File: ui/components/app/assets/asset-list/asset-list-control-bar/index.scss (hunk 4 above)
Comment (gambinish) [reply ui/components/app/assets/asset-list/asset-list-control-bar/index.scss]: https://github.com/MetaMask/metamask-extension/pull/27084/commits/106cc431ecd16632b350a15318e83dd5b6ac5c38

**Comment by gambinish** [comment]
closing in favor of: https://github.com/MetaMask/metamask-extension/pull/27184
